storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -283,9 +346,22 @@ func (s *fileSystemSource) handleErrorsAndUnfulfilled(
 	}
 }
 
+func (s *fileSystemSource) tagsFromTagsIter(
+	iter ident.TagIterator,
+) (ident.Tags, error) {
+	tags := make(ident.Tags, 0, iter.Remaining())
```

Comment:
this code repeats all the time, do you think its worth adding a new method to the iter to just do it?

Reply:
Also we're not pooling these arrays yeah? I think I need ot do something similar for the commitlog bootstrappign stuff and wasn't sure. I thought @prateek mentioned he created a pool for it

Reply:
Yeah I don't think pooling is necessary cause these ids/tags would be retained in memory by the `series` (once we transfer ownership correctly). Longer term I think it'd be worthwhile to see if bulk allocating the slices helps perf but we can come back to that.